.circleci/config.yml

```diff
@@ -450,6 +450,113 @@ jobs:
+            equal: ["nextjs", "<< pipeline.parameters.FRAMEWORK >>"]
+          steps:
+            - run:
+                name: Validate Next.js Bundle Structure
+                working_directory: ~/project/repository/output
+                command: |
+                  echo "Checking Next.js bundle structure..."
+
+                  # Verify essential Next.js files and directories
+                  if [ ! -f "server.js" ]; then
+                    echo "Error: server.js is missing"
+                    exit 1
+                  fi
+
+                  if [ ! -d ".next" ]; then
+                    echo "Error: .next directory is missing"
+                    exit 1
+                  fi
+
+                  if [ ! -d ".next/static" ]; then
+                    echo "Error: .next/static directory is missing"
+                    exit 1
+                  fi
+
+                  if [ ! -d "public" ]; then
+                    echo "Warning: public directory is missing, this might be okay if no static assets are used"
+                  else
+                    echo "Public directory is present"
+                  fi
+
+                  echo "Next.js bundle structure is valid"
+
+            - run:
+                name: Validate Next.js Bundle Dependencies
+                working_directory: ~/project/repository/output
+                command: |
+                  echo "Checking for missing dependencies in the Next.js bundle..."
+
+                  # Check if package.json exists and node_modules is populated
+                  if [ ! -d "node_modules" ] || [ $(find node_modules -mindepth 1 -maxdepth 1 | wc -l) -eq 0 ]; then
+                    echo "Error: node_modules directory is missing or empty"
+                    exit 1
+                  fi
+
+                  # Check for common Next.js required modules
+                  required_modules=("next" "react" "react-dom")
+                  for module in "${required_modules[@]}"; do
+                    if [ ! -d "node_modules/$module" ]; then
+                      echo "Error: Required dependency '$module' is missing"
+                      exit 1
+                    fi
+                  done
+
+                  echo "All required dependencies are present"
+
+            - run:
+                name: Test Next.js Bundle
+                working_directory: ~/project/repository/output
+                command: |
+                  echo "Testing Next.js bundle to verify it can start properly..."
+
+                  # Install curl for HTTP testing
+                  sudo apt-get update
+                  sudo apt-get install -y curl
+
+                  # Start the server with specific host binding
+                  PORT=3000
+                  NODE_OPTIONS="--max-old-space-size=4096" HOST="0.0.0.0" node server.js &
+                  server_pid=$!
+
+                  # Wait for server to start
+                  echo "Waiting for server to start..."
+                  sleep 15
```

Comment:
we could poll instead of sleep. like wait one second and try the request, retrying up to 15 times